actors/organizer.go

```diff
@@ -314,79 +405,364 @@ func (o *Organizer) handleCreatePoll(msg message.Message, canal string, query me
+	queryStr []byte,
+	err error,
+){
+	var eventStruct interface{}
+
+		switch dataToSign["object"] {
+		case "lao":
+			laoData, err := parser.ParseDataCreateLAO(toSignStruct.Data)
+			if err != nil {
+				log.Printf("unable to parse stored LAO infos in handleWitnessMessage()")
+				return nil, err
+			}
+
+			eventStruct = event.LAO{
+				ID:            string(laoData.ID),
+				Name:          laoData.Name,
+				Creation:      laoData.Creation,
+				OrganizerPKey: string(laoData.Organizer),
+				Witnesses:     lib.NestedByteArrayToStringArray(laoData.Witnesses),
+			}
+			queryStr, err = parser.ComposeBroadcastStateLAO(eventStruct.(event.LAO), laoData, o.PublicKey, dataWitnessMess.Signature)
+			if err != nil {
+				log.Printf("could not compose a state update broadcast message")
+				return nil, err
+			}
+
+		default:
+			log.Printf("witness not able to witness something else than LAO state update for now")
+			return nil, lib.ErrNotYetImplemented
+
+		}
+
+		err = db.UpdateChannel(eventStruct, o.database)
+		if err != nil {
+			log.Printf("error updating the message in the database")
+			return nil, err
+		}
+
+	return queryStr,nil
 }
 
-func (o *Organizer) handleCatchup(query message.Query) ([]byte, error) {
+// handleCatchup is the function to handle a received message requesting a catchup on a channel.
+// It is called by HandleReceivedMessage, and returns the current state of a channel.
+func (o *organizer) handleCatchup(query message.Query) ([]byte, error) {
 	// TODO maybe pass userId as an arg in order to check access rights later on?
 	params, err := parser.ParseParams(query.Params)
 	if err != nil {
-		fmt.Printf("unable to analyse paramsLight in handleCatchup()")
+		log.Printf("unable to analyse params in handleCatchup()")
 		return nil, lib.ErrRequestDataInvalid
 	}
 	history := db.GetChannel([]byte(params.Channel), o.database)
 
 	return history, nil
 }
 
-//just to implement the interface, this function is not needed for the Organizer (as he's the one sending this message)
-func (o *Organizer) handleLAOState(msg message.Message, chann string, query message.Query) (message, channel []byte, err error) {
-	return nil, nil, lib.ErrInvalidAction
+//handleLAOState is just here to implement the Actor interface. It returns an error as, in the current implementation there
+// is only one organizer, and he's the one sending this message. Hence he should not be receiving it.
+func (o *organizer) handleLAOState(msg message.Message) (msgAndChannel []lib.MessageAndChannel, err error) {
+	return nil, lib.ErrInvalidAction
+}
+//handleMeetingState is just here to implement the Actor interface. It returns an error as, in the current implementation there
+// is only one organizer, and he's the one sending this message. Hence he should not be receiving it.
+func (o *organizer) handleMeetingState(msg message.Message) (msgAndChannel []lib.MessageAndChannel, err error) {
+	return nil, lib.ErrInvalidAction
+}
```

Comment:
will this really be in the protocol one day ? And as it's not the case right now I'd prefer not having it and returning an error notImplementedYet... Your thoughts ?